packages/emotion/native/Css.ml

```diff
@@ -298,40 +300,67 @@ let keyframes_to_string keyframes =
   in
   keyframes |> List.map pp_keyframe |> String.concat ""
 
-let render_hash prefix hash styles =
+let render_hash hash styles =
   let is_label = function D ("label", value) -> Some value | _ -> None in
   match List.find_map is_label styles with
-  | None -> Printf.sprintf "%s-%s" prefix hash
-  | Some label -> Printf.sprintf "%s-%s-%s" prefix hash label
+  | None -> Printf.sprintf "%s" hash
+  | Some label -> Printf.sprintf "%s-%s" hash label
 
 let style (styles : rule list) =
   match styles with
   | [] -> ""
   | _ ->
+    let preffix = "css" in
     let hash = Murmur2.default (rules_to_string styles) in
-    let className = render_hash "css" hash styles in
-    Stylesheet.push instance (className, Classnames styles);
-    className
+    let className = render_hash hash styles in
+    Stylesheet.push instance (preffix, className, Classnames styles);
+    Printf.sprintf "%s-%s" preffix className
+
+let global (styles : rule list) =
+  match styles with
+  | [] -> ""
+  | _ ->
+    let hash = Murmur2.default (rules_to_string styles) in
+    Stylesheet.push instance ("", hash, Global styles);
```

Comment:
Looking at this I wonder if the prefix can be stored inside Classnames and don't need to do anything in Global, thoughts?

Reply:
Nice idea, what about something like this:

```ocaml
let style (styles : rule list) =
  match styles with
  | [] -> ""
  | _ ->
    let hash = render_hash (Murmur2.default (rules_to_string styles)) styles in
    let className = Printf.sprintf "%s-%s" "css" hash in
    Stylesheet.push instance (hash, Classnames (className, styles));
    className

let global (styles : rule list) =
  match styles with
  | [] -> ""
  | _ ->
    let hash = Murmur2.default (rules_to_string styles) in
    Stylesheet.push instance (hash, Global styles);
    hash

let keyframes (keyframes : (int * rule list) list) =
  match keyframes with
  | [] -> ""
  | _ ->
    let hash = Murmur2.default (keyframes_to_string keyframes) in
    let animationName = Printf.sprintf "%s-%s" "animation" hash in
    Stylesheet.push instance (hash, Keyframes (animationName, keyframes));
    animationName
```

Then we can manage hash, which is a value for all styles, and the variant provides the need for extra info.

The usage can be:

```ocaml
let render_style_tag () =
  Stylesheet.get_all instance
  |> List.fold_left
       (fun accumulator (_, rules) ->
         match rules with
         | Global rules ->
           Printf.sprintf "%s %s" accumulator (rules_to_string rules)
         | Classnames (className, rules) ->
           let rules = pp_rules className rules |> String.trim in
           Printf.sprintf "%s %s" accumulator rules
         | Keyframes (animationName, keyframes) ->
           let rules = pp_keyframes animationName keyframes |> String.trim in
           Printf.sprintf "%s %s" accumulator rules)
       ""
  |> String.trim
```